### PR TITLE
Adding leafrefResolver to openapi spec

### PR DESCRIPTION
**Pull request by frouzbeh**

### CODE REVIEW

File: models/sdn-fabric-0.1.x/openapi.yaml [hunk 1]
```diff
@@ -248,6 +248,7 @@ components:
           title: model-id
           type: string
           x-leafref: /sm:switch-model/sm:switch-model-id
+          x-leafrefResolver: /sdn-fabric/v0.1.x/{fabric-id}/switch/{switch-id}/model-id/values
```
Comment (frouzbeh) [inline models/sdn-fabric-0.1.x/openapi.yaml:251]: @SeanCondon The PR is not done yet, I just wanted to make sure I got the path right as we discussed. Please check and let me know.

**Review by teone — COMMENTED** [review]
@frouzbeh the path looks good to me

**Comment by frouzbeh** [comment]
> @frouzbeh the path looks good to me

@teone  Thanks, did you check all paths? Because for the leaflists I'm adding the id at the end. Like what we did for the tagged-vlan

**Comment by teone** [comment]
Yes, all of these make sense to me, would be probably good to generate the new `openapi` spec for the Aether models too, so we can see what a `leafref` to a list looks like (I'm not sure there is such thing, as you'd probably reference a specific attribute of list elements)

File: models/sdn-fabric-0.1.x/openapi.yaml (hunk 1 above)
Comment (SeanCondon) [reply models/sdn-fabric-0.1.x/openapi.yaml:251]: this is right

**Comment by frouzbeh** [comment]
@teone @SeanCondon Please review the patch. I will revert back the sd-fabric api changes after review.

File: models/sdn-fabric-0.1.x/openapi.yaml [hunk 2]
```diff
@@ -167,6 +167,22 @@ components:
       x-keys:
       - dhcp-server-id
       x-list-multiple: true
+    LeafRefResolver:
```
Comment (SeanCondon) [inline models/sdn-fabric-0.1.x/openapi.yaml:170]: looks good - a better name might be `LeafRefOptions`

**Review by SeanCondon — APPROVED** [review]
you will have to make the update of `models/*` in a different PR to the `pkg` ones

File: models/sdn-fabric-0.1.x/openapi.yaml (hunk 2 above)
Comment (frouzbeh) [reply models/sdn-fabric-0.1.x/openapi.yaml:170]: @SeanCondon Yes, I will create a separate PR to update the models. sd-fabric here is just a temporary example.

**Pull request merged**